subprojects/core/src/test/groovy/org/gradle/api/internal/collections/ElementSourceSpec.groovy

```diff
@@ -22,9 +22,10 @@ import org.gradle.api.internal.provider.ChangingValue
 import org.gradle.api.internal.provider.ChangingValueHandler
 import org.gradle.api.internal.provider.CollectionProviderInternal
 import org.gradle.api.internal.provider.ProviderInternal
-import spock.lang.Specification
 
-abstract class AbstractElementSourceTest extends Specification {
+abstract class ElementSourceSpec extends PendingSourceSpec {
```

Comment:
```suggestion
abstract class AbstractElementSourceSpec extends PendingSourceSpec {
```

I'd keep the abstract, though I don't feel strongly about it.

Reply:
I would rather keep it as `ElementSourceSpec`, because it corresponds to the `ElementSource` interface.
This is similar to how we have `ProviderSpec`, `PropertySpec`, etc for corresponding interfaces.